content/Projects/Observability/Platform/faq.md

```diff
@@ -0,0 +1,138 @@
+# Debugging the in-cluster monitoring stack
+
+## Overview
+
+This serves as a collection of resources that relate to FAQ around configuring/debugging the in-cluster monitoring stack.
+
+## Identify the cause of some common issues
+
+### ServiceMonitor created but no metrics from the service
+
+The monitoring stack relies on the `ServiceMonitor` custom resource in order to tell Prometheus what endpoints to scrape.
+
+A detailed description of how the resources are linked exists [here](https://github.com/prometheus-operator/prometheus-operator/blob/c6b5cc2e178002683ab3afe8d023a8bb2508ba19/Documentation/troubleshooting.md#overview-of-servicemonitor-tagging-and-related-elements), but we will walk through some common issues to debug the case of missing metrics.
+
+1. Ensure the `serviceMonitorSelector` in the `Prometheus` CR matches the key in the `ServiceMonitor` labels.
+2. The `Service` you want to scrape *must* have an explicitly named port.
+3. The `ServiceMonitor` *must* reference the `port` by this name.
+
+Assuming this criteria is met but the metrics don't exist, we can try debug the following.
+
+```bash
+oc -n openshift-monitoring get secret prometheus-k8s -o json | jq -r '.data["prometheus.yaml.gz"]' | base64 -d | gunzip | grep "<service-monitor-name>"
```

Comment:
another option is to check the running config using the Prometheus `/api/v1/status/config` endpoint.